packages/cx-elements/doc/Design/CanvasxTokens/TokenDomains.md

```diff
@@ -0,0 +1,153 @@
+  eg, node keys.
+
+### Special Domains
+
+The following special domains have values which are not extended domain
+objects:
+
+- `Compose`: This special domain allows specifying other named tokens which
+  should be composed with this one. See
+  [Composition and Extension](../CanvasxTokens#composition-and-extension) for
+  more details. Example:
+  ```ts
+  const Sticky = asHeaderToken({
+    ...Base,
+    Compose: {
+      // Apply the WithSticky token to the component as a whole.
+      WithSticky,
+    },
+  })
+  ```
+- `Flow`: By default, the tokens HOC's for each domain are composed using
+  the Bodiless `asToken` utility.  This behavior can be overridden by specifying
+  a different composer here.  This is useful for wrapping the components in a
+  condition (using `flowIf`).
+  ```ts
+  const WithBlueIfActive = asTokenSpec()({
+    Core: {
+      _: 'text-blue',
+    },
+    Flow: flowIf(({ isActive }) => Boolean(isActive)),
+  }),
+  ```
+- `Meta`: This Domain takes a token metadata object which should be attached to
+  the token (and to any component to which the token is applied).
+  ```ts
+  const WithBlueBackground = asTokenSpec()({
+    Core: {
+      _: 'bg-blue',
+    },
+    Meta: asToken.meta.term('Color')('Blue'),
+  });
+  ```
+
+## Order ot domains
+
+HOC's defined for each domain are applied in a fixed order (as listed above),
+regardless of the order in which they are specified in an individual token.
+
+For example, given
+
+```ts
+const Token = asTokenSpec()({
+  Theme: {
+    _: 'bar',
+  },
+  Layout: {
+    _: 'baz',
+  },
+  Core: {
+    _: 'foo',
+  },
+});
+```
+
+then
+
+```ts
+as(Token)
+```
+
+is the same as
+
+```ts
+as(
+  pick(Token, 'Core'),
+  pick(Token, 'Layout'),
+  pick(Token, 'Theme'),
+);
+```
+## Extended Design
+
+With exception of the three special domains ('Compose', 'Meta' and 'Flow'),
+the value of each domain is an extended design object. This is similar
+to a normal Bodiless design object, with the following differences:
+- The value of each key can by any of the following:
+  - A normal Bodiless token  HOC (as in a normal design object)
+  - A token specification object (which will be converted to an HOC by
+    being wrapped in `as`).
+  - A string (which will be interpreted as a set of CSS classes to be
+   added to the component via `addClasses`).
+- There is a special `_` key which defines tokens/hoc's which should be
+  applied to the component as a whole.
+
+Example:
+
+```ts
+const Example = asLinkToken({
+  Core: {
+    // Apply HOC to the component as a whole
+    _: addProps({
+      id: 'special-link-id',
+    }),
+    // Apply a token to the Body key.
+    Body: cxElements.WithLinkStyles,
+    // Add a class to the icon
+    ExternalIcon: 'text-xl',
+  },
+});
+```
+## Additional Considerations and Gotchas
+
+### Access to component data
+Note the position of hte `Editors` and `Schema` domains which usually provide
```

Comment:
```suggestion
Note the position of the `Editors` and `Schema` domains which usually provide
```

Reply:
@hvanyo ,

I'll merge the PR and address it directly in my branch. Thanks!